src/XMakeBuildEngine/Construction/ProjectRootElement.cs

```diff
@@ -1714,8 +1714,23 @@ public void Save(Encoding saveEncoding)
                 {
                     using (ProjectWriter projectWriter = new ProjectWriter(_projectFileLocation.File, saveEncoding))
                     {
-                        projectWriter.Initialize(XmlDocument);
-                        XmlDocument.Save(projectWriter);
+                        var xmlWithNoImplicits = (XmlDocument)XmlDocument.Clone();
+
+                        var implicitElements =
+                            xmlWithNoImplicits.SelectNodes($"//*[@{XMakeAttributes.@implicit}]");
+
+                        if (implicitElements == null)
```

Comment:
I can't think of a reason for this to fail either, but it seemed worth checking and throwing. Would you rather I just let a NRE happen later?